eventuals/task.h

```diff
@@ -360,25 +394,35 @@ struct _TaskFromToWith {
           std::move(dispatch_)};
     }
 
-    Callback<
-        Action,
-        std::optional<std::exception_ptr>&&,
-        Args_&&...,
-        // Can't have a 'void' argument type so we are using 'std::monostate'.
-        std::optional<
-            std::conditional_t<
-                std::is_void_v<From_>,
-                std::monostate,
-                From_>>&&,
-        std::unique_ptr<void, Callback<void*>>&,
-        Interrupt&,
+    // The 'dispatch_' might be both function that creates an eventual
+    // or argument that should be passed throw 'Task' in the 'Start' method
+    // of 'Task::Continuation' used in 'Task::Success'.
+    std::variant<
         std::conditional_t<
             std::is_void_v<To_>,
-            Callback<>&&,
-            Callback<To_>&&>,
-        Callback<std::exception_ptr>&&,
-        Callback<>&&>
+            std::monostate,
+            std::decay_t<To_>>,
```

Comment:
Why do you need the `std::decay_t` here and above? If you don't actually need it please remove otherwise please leave a comment!

Reply:
When `To_` is a reference it can't be stored at `std::variant`

variant:1338:7 error: static_assert failed due to requirement - "variant must have no reference alternative"

Reply:
Let's please add a test for this case!